opengrok-indexer/src/main/java/org/opengrok/indexer/util/BeanBuilder.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ */
+
+package org.opengrok.indexer.util;
+
+import net.sf.cglib.beans.BeanGenerator;
```

Comment:
I had eyeballed [ModelMapper](http://modelmapper.org/getting-started/) for possible usefulness, and I see on the [cglib wiki](https://github.com/cglib/cglib/wiki) that ModelMapper uses cglib. If you get a chance, I'm curious what you think.

Reply:
No matter how I tried to use ModelMapper it was always returning the DTO object with all members being null. Then I realized its major downside - it assumes getters and setters are already in place in the DTO object.

Reply:
In the end I combined modelmapper with cglib and annotations to achieve the desired effect.